tests/test_threadpoolctl.py

```diff
@@ -101,6 +101,26 @@ def test_set_threadpool_limits_by_api(user_api):
     assert threadpool_info() == original_infos
 
 
+def test_threadpool_limits_function_with_side_effect():
+    # Check that threadpool_limits can be used as a function with
+    # side effects instead of a context manager.
+    original_infos = threadpool_info()
+
+    threadpool_limits(limits=1)
```

Comment:
You could also use `limitation = threadpool_limits(limits=1)` and then use `limitation.unregister()` in the finally to avoid side effects.

Reply:
Actually, do we really want to make `unregister` public?

Reply:
I added a test for this in a6acd86.